### ISSUE TITLE
bug(gatsby-source-strapi): Gatsby does not pick up latest article

### ISSUE THREAD

**Issue by fgroenendijk** [gatsby-uc/plugins#393]
## Describe the bug

On my system there are more than 250 articles, and because of pagination the latest just saved article is not picked up by gatsby.

## To Reproduce

1. `gatsby run build`
2. save an article in strapi to update the article timestamp
3. `gatsby run build`

The second `gatsby run build` does not pick up the latest article.

The output is: 

`info Starting to fetch data from Strapi - /api/articles with {"pagination":{"pageSize":250,"page":1},"populate":"*"}
info Starting to fetch data from Strapi - /api/articles with "pagination[pageSize]=250&pagination[page]=2&populate=%2A"
...
info Starting to fetch data from Strapi - /api/articles with "pagination[pageSize]=250&pagination[page]=21&populate=%2A"
info Starting to fetch data from Strapi - /api/articles with {"pagination":{"pageSize":250,"page":21},"populate":"*","filters":{"updatedAt":{"$gt":1678290635737}}}`

So my observation is that it goes to page 21 with a pagesize of 250 and tries to get the latest article.

## Expected behavior

Gatsby picking up the latest just saved article.

Basically I've tried removing the pagination when using the gatsby cache, but that seemed to break gatsby.

PR: https://github.com/gatsby-uc/plugins/pull/394

**Comment by laurenskling** [comment]
Very interesting. I'll look into it.

**Comment by fgroenendijk** [comment]
Thanks!

**Comment by laurenskling** [comment]
I actually never went over the default 250 pageLimit myself (: 
I also see the log acts a little different on paged requests. I like that one better. (the actual qs.stringify result)

The problem lies in this line: https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-source-strapi/src/fetch.js#L142
this is not how spread works for nested objects :( 
Will try to make a fix tomorrow. Including some fixes to the logging. I would prefer to make all logs the qs.stringify version. So you can actually see how the REST endpoint is requested. Do you agree?

**Comment by laurenskling** [comment]
Glad to hear that :)
